File: py/cache_constants.py
"""
缓存键常量类 - Python版本
与Java端保持一致的缓存键命名规范

主要功能：
- 统一的缓存键前缀和命名规范
- 缓存过期时间常量
- 缓存键构建工具方法

版本: 1.0.0
"""

class CacheConstants:
    """缓存键常量类"""
    
    # ================================ 缓存键前缀 ================================
    
    CACHE_PREFIX = "poetize:"
    
    # ================================ 用户相关缓存 ================================
    
    USER_CACHE_PREFIX = CACHE_PREFIX + "user:"
    USER_SESSION_PREFIX = CACHE_PREFIX + "session:"
    LOGIN_FAIL_PREFIX = CACHE_PREFIX + "login:fail:"
    
    # ================================ 文章相关缓存 ================================
    
    ARTICLE_CACHE_PREFIX = CACHE_PREFIX + "article:"
    ARTICLE_LIST_PREFIX = CACHE_PREFIX + "article:list:"
    HOT_ARTICLES_KEY = CACHE_PREFIX + "article:hot"
    ARTICLE_VIEW_PREFIX = CACHE_PREFIX + "article:view:"
    
    # ================================ 评论相关缓存 ================================
    
    COMMENT_LIST_PREFIX = CACHE_PREFIX + "comment:list:"
    COMMENT_COUNT_PREFIX = CACHE_PREFIX + "comment:count:"
    
    # ================================ 分类标签缓存 ================================
    
    SORT_LIST_KEY = CACHE_PREFIX + "sort:list"
    LABEL_LIST_PREFIX = CACHE_PREFIX + "label:list:"
    
    # ================================ 系统配置缓存 ================================
    
    WEB_INFO_KEY = CACHE_PREFIX + "webinfo"
    SYS_CONFIG_PREFIX = CACHE_PREFIX + "config:"
    
    # ================================ 安全相关缓存 ================================
    
    IP_ATTACK_PREFIX = CACHE_PREFIX + "security:attack:"
    IP_BLACKLIST_PREFIX = CACHE_PREFIX + "security:blacklist:"
    CAPTCHA_PREFIX = CACHE_PREFIX + "captcha:"
    
    # ================================ 统计相关缓存 ================================
    
    VISIT_STATS_PREFIX = CACHE_PREFIX + "stats:visit:"
    ONLINE_USERS_KEY = CACHE_PREFIX + "stats:online"
    
    # ================================ 第三方服务缓存 ================================
    
    OAUTH_STATE_PREFIX = CACHE_PREFIX + "oauth:state:"
    TRANSLATE_PREFIX = CACHE_PREFIX + "translate:"
    
    # ================================ Python特有缓存 ================================

    # 邮件配置缓存
    EMAIL_CONFIG_KEY = CACHE_PREFIX + "email:config"
    EMAIL_TEST_PREFIX = CACHE_PREFIX + "email:test:"

    # 验证码配置缓存
    CAPTCHA_CONFIG_KEY = CACHE_PREFIX + "captcha:config"
    CAPTCHA_PUBLIC_CONFIG_KEY = CACHE_PREFIX + "captcha:public:config"

    # SEO配置缓存
    SEO_CONFIG_KEY = CACHE_PREFIX + "seo:config"
    SEO_SITEMAP_KEY = CACHE_PREFIX + "seo:sitemap"
    SEO_ROBOTS_KEY = CACHE_PREFIX + "seo:robots"
    SEO_AI_ANALYSIS_PREFIX = CACHE_PREFIX + "seo:ai:analysis:"
    SEO_PUSH_STATUS_PREFIX = CACHE_PREFIX + "seo:push:status:"

    # AI翻译缓存
    AI_TRANSLATE_PREFIX = CACHE_PREFIX + "ai:translate:"
    AI_CONFIG_KEY = CACHE_PREFIX + "ai:config"

    # AI聊天缓存
    AI_CHAT_CONFIG_KEY = CACHE_PREFIX + "ai:chat:config"
    AI_CHAT_RESPONSE_PREFIX = CACHE_PREFIX + "ai:chat:response:"
    AI_CHAT_CONVERSATION_PREFIX = CACHE_PREFIX + "ai:chat:conversation:"

    # Web管理缓存
    WEB_INFO_KEY = CACHE_PREFIX + "web:info"
    WEB_INFO_ADMIN_KEY = CACHE_PREFIX + "web:info:admin"
    WEB_INFO_DETAILS_PREFIX = CACHE_PREFIX + "web:info:details:"

    # 访问统计缓存
    VISIT_STATS_DAILY_PREFIX = CACHE_PREFIX + "stats:daily:"
    VISIT_STATS_SUMMARY_PREFIX = CACHE_PREFIX + "stats:summary:"
    
    # 访问统计缓存
    VISIT_COUNT_PREFIX = CACHE_PREFIX + "visit:count:"
    VISIT_IP_PREFIX = CACHE_PREFIX + "visit:ip:"
    
    # ================================ 缓存过期时间常量（秒） ================================
    
    # 基础过期时间
    DEFAULT_EXPIRE_TIME = 600        # 10分钟
    SHORT_EXPIRE_TIME = 300          # 5分钟
    LONG_EXPIRE_TIME = 3600          # 1小时
    VERY_LONG_EXPIRE_TIME = 86400    # 24小时
    
    # 特定功能过期时间
    SESSION_EXPIRE_TIME = 604800     # 7天
    CAPTCHA_EXPIRE_TIME = 300        # 5分钟
    IP_ATTACK_EXPIRE_TIME = 3600     # 1小时
    IP_BLACKLIST_EXPIRE_TIME = 86400 # 24小时
    
    # OAuth相关过期时间
    OAUTH_STATE_EXPIRE_TIME = 600    # 10分钟
    OAUTH_TOKEN_EXPIRE_TIME = 3600   # 1小时
    
    # 翻译缓存过期时间
    TRANSLATE_EXPIRE_TIME = 86400    # 24小时
    
    # 配置缓存过期时间
    CONFIG_EXPIRE_TIME = 3600        # 1小时
    
    # 统计数据过期时间
    STATS_EXPIRE_TIME = 1800         # 30分钟

    # Python API特定过期时间
    # ================================ 持久化缓存（配置类数据） ================================
    # 这些配置变更频率极低，使用持久化缓存，只在配置更新时主动清理
    EMAIL_CONFIG_EXPIRE_TIME = None          # 邮件配置 - 持久化缓存
    CAPTCHA_CONFIG_EXPIRE_TIME = None        # 验证码配置 - 持久化缓存
    SEO_CONFIG_EXPIRE_TIME = None            # SEO配置 - 持久化缓存
    AI_CHAT_CONFIG_EXPIRE_TIME = None        # AI聊天配置 - 持久化缓存
    WEB_INFO_EXPIRE_TIME = None              # 网站信息 - 持久化缓存
    WEB_INFO_ADMIN_EXPIRE_TIME = None        # 管理员网站信息 - 持久化缓存

    # ================================ 临时缓存（动态数据） ================================
    # 这些数据需要定期刷新或可能频繁变化，保留TTL
    SEO_SITEMAP_EXPIRE_TIME = 1800           # sitemap 30分钟（内容可能变化）
    SEO_ROBOTS_EXPIRE_TIME = 1800            # robots.txt 30分钟（内容可能变化）
    SEO_AI_ANALYSIS_EXPIRE_TIME = 7200       # SEO AI分析2小时
    SEO_PUSH_STATUS_EXPIRE_TIME = 300        # SEO推送状态5分钟
    AI_CHAT_RESPONSE_EXPIRE_TIME = 3600      # AI聊天响应1小时
    VISIT_STATS_EXPIRE_TIME = 300            # 访问统计5分钟
    
    # ================================ 工具方法 ================================
    
    @staticmethod
    def build_user_key(user_id: int) -> str:
        """构建用户缓存键"""
        return f"{CacheConstants.USER_CACHE_PREFIX}{user_id}"
    
    @staticmethod
    def build_session_key(token: str) -> str:
        """构建用户会话缓存键"""
        return f"{CacheConstants.USER_SESSION_PREFIX}{token}"
    
    @staticmethod
    def build_article_key(article_id: int) -> str:
        """构建文章缓存键"""
        return f"{CacheConstants.ARTICLE_CACHE_PREFIX}{article_id}"
    
    @staticmethod
    def build_article_list_key(sort_id: int, page: int, size: int) -> str:
        """构建文章列表缓存键"""
        return f"{CacheConstants.ARTICLE_LIST_PREFIX}{sort_id}:{page}:{size}"
    
    @staticmethod
    def build_comment_list_key(source: int, comment_type: str) -> str:
        """构建评论列表缓存键"""
        return f"{CacheConstants.COMMENT_LIST_PREFIX}{source}:{comment_type}"
    
    @staticmethod
    def build_ip_attack_key(ip: str) -> str:
        """构建IP攻击缓存键"""
        return f"{CacheConstants.IP_ATTACK_PREFIX}{ip}"
    
    @staticmethod
    def build_ip_blacklist_key(ip: str) -> str:
        """构建IP黑名单缓存键"""
        return f"{CacheConstants.IP_BLACKLIST_PREFIX}{ip}"
    
    @staticmethod
    def build_sys_config_key(config_key: str) -> str:
        """构建系统配置缓存键"""
        return f"{CacheConstants.SYS_CONFIG_PREFIX}{config_key}"
    
    @staticmethod
    def build_oauth_state_key(state: str) -> str:
        """构建OAuth状态缓存键"""
        return f"{CacheConstants.OAUTH_STATE_PREFIX}{state}"
    
    @staticmethod
    def build_translate_key(text_hash: str) -> str:
        """构建翻译缓存键"""
        return f"{CacheConstants.TRANSLATE_PREFIX}{text_hash}"
    
    @staticmethod
    def build_captcha_key(session_id: str) -> str:
        """构建验证码缓存键"""
        return f"{CacheConstants.CAPTCHA_PREFIX}{session_id}"
    
    @staticmethod
    def build_email_test_key(config_hash: str) -> str:
        """构建邮件测试缓存键"""
        return f"{CacheConstants.EMAIL_TEST_PREFIX}{config_hash}"
    
    @staticmethod
    def build_visit_count_key(date: str) -> str:
        """构建访问统计缓存键"""
        return f"{CacheConstants.VISIT_COUNT_PREFIX}{date}"
    
    @staticmethod
    def build_visit_ip_key(ip: str, date: str) -> str:
        """构建访问IP缓存键"""
        return f"{CacheConstants.VISIT_IP_PREFIX}{ip}:{date}"
    
    @staticmethod
    def build_ai_translate_key(text_hash: str, source_lang: str, target_lang: str) -> str:
        """构建AI翻译缓存键"""
        return f"{CacheConstants.AI_TRANSLATE_PREFIX}{text_hash}:{source_lang}:{target_lang}"

    # ================================ Python API缓存键构建方法 ================================

    @staticmethod
    def build_web_info_details_key(auth_token_hash: str) -> str:
        """构建网站详细信息缓存键"""
        return f"{CacheConstants.WEB_INFO_DETAILS_PREFIX}{auth_token_hash}"

    @staticmethod
    def build_seo_ai_analysis_key(site_info_hash: str) -> str:
        """构建SEO AI分析缓存键"""
        return f"{CacheConstants.SEO_AI_ANALYSIS_PREFIX}{site_info_hash}"

    @staticmethod
    def build_seo_push_status_key(url_hash: str, engine: str) -> str:
        """构建SEO推送状态缓存键"""
        return f"{CacheConstants.SEO_PUSH_STATUS_PREFIX}{engine}:{url_hash}"

    @staticmethod
    def build_ai_chat_response_key(message_hash: str, config_hash: str) -> str:
        """构建AI聊天响应缓存键"""
        return f"{CacheConstants.AI_CHAT_RESPONSE_PREFIX}{config_hash}:{message_hash}"

    @staticmethod
    def build_ai_chat_conversation_key(conversation_id: str, user_id: str) -> str:
        """构建AI聊天会话缓存键"""
        return f"{CacheConstants.AI_CHAT_CONVERSATION_PREFIX}{user_id}:{conversation_id}"

    @staticmethod
    def build_visit_stats_daily_key(days: int) -> str:
        """构建每日访问统计缓存键"""
        return f"{CacheConstants.VISIT_STATS_DAILY_PREFIX}{days}"

    @staticmethod
    def build_visit_stats_summary_key(period: str) -> str:
        """构建访问统计摘要缓存键"""
        return f"{CacheConstants.VISIT_STATS_SUMMARY_PREFIX}{period}"

# 缓存键模式常量
class CachePatterns:
    """缓存键模式常量"""
    
    # 用户相关模式
    USER_PATTERN = CacheConstants.USER_CACHE_PREFIX + "*"
    SESSION_PATTERN = CacheConstants.USER_SESSION_PREFIX + "*"
    
    # 文章相关模式
    ARTICLE_PATTERN = CacheConstants.ARTICLE_CACHE_PREFIX + "*"
    ARTICLE_LIST_PATTERN = CacheConstants.ARTICLE_LIST_PREFIX + "*"
    
    # 安全相关模式
    IP_ATTACK_PATTERN = CacheConstants.IP_ATTACK_PREFIX + "*"
    IP_BLACKLIST_PATTERN = CacheConstants.IP_BLACKLIST_PREFIX + "*"
    
    # OAuth相关模式
    OAUTH_STATE_PATTERN = CacheConstants.OAUTH_STATE_PREFIX + "*"
    
    # 翻译相关模式
    TRANSLATE_PATTERN = CacheConstants.TRANSLATE_PREFIX + "*"
    AI_TRANSLATE_PATTERN = CacheConstants.AI_TRANSLATE_PREFIX + "*"

# 缓存分组常量
class CacheGroups:
    """缓存分组常量"""
    
    USER_GROUP = "user"
    ARTICLE_GROUP = "article"
    COMMENT_GROUP = "comment"
    SECURITY_GROUP = "security"
    CONFIG_GROUP = "config"
    STATS_GROUP = "stats"
    OAUTH_GROUP = "oauth"
    TRANSLATE_GROUP = "translate"
    EMAIL_GROUP = "email"
    CAPTCHA_GROUP = "captcha"
    SEO_GROUP = "seo"

File: poetize-server/poetry-web/src/main/java/com/ld/poetry/constants/CacheConstants.java
package com.ld.poetry.constants;

/**
 * 缓存键常量类
 * 统一管理Redis缓存键的命名规范
 * 
 * @author LeapYa
 * @since 2025-07-20
 */
public class CacheConstants {

    /**
     * 缓存键前缀
     */
    public static final String CACHE_PREFIX = "poetize:";

    // ================================ 用户相关缓存 ================================
    
    /**
     * 用户信息缓存键前缀
     * 格式: poetize:user:{userId}
     */
    public static final String USER_CACHE_PREFIX = CACHE_PREFIX + "user:";
    
    /**
     * 用户会话缓存键前缀
     * 格式: poetize:session:{token}
     */
    public static final String USER_SESSION_PREFIX = CACHE_PREFIX + "session:";
    
    /**
     * 用户登录失败次数缓存键前缀
     * 格式: poetize:login:fail:{username}
     */
    public static final String LOGIN_FAIL_PREFIX = CACHE_PREFIX + "login:fail:";

    // ================================ 文章相关缓存 ================================
    
    /**
     * 文章信息缓存键前缀
     * 格式: poetize:article:{articleId}
     */
    public static final String ARTICLE_CACHE_PREFIX = CACHE_PREFIX + "article:";
    
    /**
     * 文章列表缓存键前缀
     * 格式: poetize:article:list:{sortId}:{page}:{size}
     */
    public static final String ARTICLE_LIST_PREFIX = CACHE_PREFIX + "article:list:";
    
    /**
     * 热门文章缓存键
     */
    public static final String HOT_ARTICLES_KEY = CACHE_PREFIX + "article:hot";
    
    /**
     * 文章浏览量缓存键前缀
     * 格式: poetize:article:view:{articleId}
     */
    public static final String ARTICLE_VIEW_PREFIX = CACHE_PREFIX + "article:view:";

    /**
     * 用户文章列表缓存键前缀
     * 格式: poetize:user:article:list:{userId}
     */
    public static final String USER_ARTICLE_LIST_PREFIX = CACHE_PREFIX + "user:article:list:";

    /**
     * 文章搜索结果缓存键前缀
     * 格式: poetize:search:article:{hashCode}
     */
    public static final String SEARCH_ARTICLE_PREFIX = CACHE_PREFIX + "search:article:";

    // ================================ 评论相关缓存 ================================
    
    /**
     * 评论列表缓存键前缀
     * 格式: poetize:comment:list:{source}:{type}
     */
    public static final String COMMENT_LIST_PREFIX = CACHE_PREFIX + "comment:list:";
    
    /**
     * 评论数量缓存键前缀
     * 格式: poetize:comment:count:{source}:{type}
     */
    public static final String COMMENT_COUNT_PREFIX = CACHE_PREFIX + "comment:count:";

    // ================================ 分类标签缓存 ================================
    


    /**
     * 分类文章列表缓存键
     */
    public static final String SORT_ARTICLE_LIST_KEY = CACHE_PREFIX + "sort:article:list";

    /**
     * 标签信息缓存键前缀
     * 格式: poetize:label:list:{sortId}
     */
    public static final String LABEL_LIST_PREFIX = CACHE_PREFIX + "label:list:";

    // ================================ 系统配置缓存 ================================
    
    /**
     * 网站信息缓存键
     * 注意：网站信息使用永久缓存（不设置过期时间）
     * @see com.ld.poetry.service.CacheService#cacheWebInfo(WebInfo)
     */
    public static final String WEB_INFO_KEY = CACHE_PREFIX + "webinfo";

    /**
     * 系统配置缓存键前缀
     * 格式: poetize:config:{configKey}
     */
    public static final String SYS_CONFIG_PREFIX = CACHE_PREFIX + "config:";

    /**
     * 管理员用户缓存键
     */
    public static final String ADMIN_CACHE_KEY = CACHE_PREFIX + "admin";

    /**
     * 点赞用户列表缓存键
     */
    public static final String ADMIRE_LIST_KEY = CACHE_PREFIX + "admire:list";

    /**
     * 家庭成员列表缓存键
     */
    public static final String FAMILY_LIST_KEY = CACHE_PREFIX + "family:list";

    // ================================ 安全相关缓存 ================================
    
    /**
     * IP攻击次数缓存键前缀
     * 格式: poetize:security:attack:{ip}
     */
    public static final String IP_ATTACK_PREFIX = CACHE_PREFIX + "security:attack:";
    
    /**
     * IP黑名单缓存键前缀
     * 格式: poetize:security:blacklist:{ip}
     */
    public static final String IP_BLACKLIST_PREFIX = CACHE_PREFIX + "security:blacklist:";
    
    /**
     * 验证码缓存键前缀
     * 格式: poetize:captcha:{sessionId}
     */
    public static final String CAPTCHA_PREFIX = CACHE_PREFIX + "captcha:";

    /**
     * 用户验证码缓存键前缀
     * 格式: poetize:user:code:{userId}:{place}:{flag}
     */
    public static final String USER_CODE_PREFIX = CACHE_PREFIX + "user:code:";

    /**
     * 邮箱验证码发送次数缓存键前缀
     * 格式: poetize:code:mail:count:{email}
     */
    public static final String CODE_MAIL_COUNT_PREFIX = CACHE_PREFIX + "code:mail:count:";

    /**
     * 忘记密码验证码缓存键前缀
     * 格式: poetize:forget:password:{contact}:{flag}
     */
    public static final String FORGET_PASSWORD_PREFIX = CACHE_PREFIX + "forget:password:";

    /**
     * 登录失败尝试缓存键前缀
     * 格式: poetize:login:attempt:{account}
     */
    public static final String LOGIN_ATTEMPT_PREFIX = CACHE_PREFIX + "login:attempt:";

    /**
     * 用户保存频率限制缓存键前缀
     * 格式: poetize:save:count:user:{userId}
     */
    public static final String SAVE_COUNT_USER_PREFIX = CACHE_PREFIX + "save:count:user:";

    /**
     * IP保存频率限制缓存键前缀
     * 格式: poetize:save:count:ip:{ip}
     */
    public static final String SAVE_COUNT_IP_PREFIX = CACHE_PREFIX + "save:count:ip:";

    /**
     * 用户文件上传频率限制缓存键前缀
     * 格式: poetize:upload:count:user:{userId}
     */
    public static final String FILE_UPLOAD_COUNT_USER_PREFIX = CACHE_PREFIX + "upload:count:user:";

    /**
     * IP文件上传频率限制缓存键前缀
     * 格式: poetize:upload:count:ip:{ip}
     */
    public static final String FILE_UPLOAD_COUNT_IP_PREFIX = CACHE_PREFIX + "upload:count:ip:";

    /**
     * 管理员token缓存键前缀
     * 格式: poetize:admin:token:{userId}
     */
    public static final String ADMIN_TOKEN_PREFIX = CACHE_PREFIX + "admin:token:";

    /**
     * 用户token缓存键前缀
     * 格式: poetize:user:token:{userId}
     */
    public static final String USER_TOKEN_PREFIX = CACHE_PREFIX + "user:token:";

    /**
     * 管理员token间隔检查缓存键前缀
     * 格式: poetize:admin:token:interval:{userId}
     */
    public static final String ADMIN_TOKEN_INTERVAL_PREFIX = CACHE_PREFIX + "admin:token:interval:";

    /**
     * 用户token间隔检查缓存键前缀
     * 格式: poetize:user:token:interval:{userId}
     */
    public static final String USER_TOKEN_INTERVAL_PREFIX = CACHE_PREFIX + "user:token:interval:";

    // ================================ 统计相关缓存 ================================
    
    /**
     * 访问统计缓存键前缀
     * 格式: poetize:stats:visit:{date}
     */
    public static final String VISIT_STATS_PREFIX = CACHE_PREFIX + "stats:visit:";
    
    /**
     * 今日访问计数缓存键
     * 格式: poetize:visit:count:today:{date}
     */
    public static final String TODAY_VISIT_COUNT_PREFIX = CACHE_PREFIX + "visit:count:today:";
    
    /**
     * 每日访问记录缓存键前缀（Redis中存储当天的访问记录）
     * 格式: poetize:visit:records:{date}
     */
    public static final String DAILY_VISIT_RECORDS_PREFIX = CACHE_PREFIX + "visit:records:";
    
    /**
     * IP今日访问标记缓存键前缀
     * 格式: poetize:visit:ip:today:{date}:{ip}_{userId}
     */
    public static final String IP_TODAY_VISIT_PREFIX = CACHE_PREFIX + "visit:ip:today:";
    
    /**
     * 在线用户数缓存键
     */
    public static final String ONLINE_USERS_KEY = CACHE_PREFIX + "stats:online";

    /**
     * IP历史记录缓存键
     */
    public static final String IP_HISTORY_KEY = CACHE_PREFIX + "ip:history";

    /**
     * IP历史统计缓存键
     */
    public static final String IP_HISTORY_STATS_KEY = CACHE_PREFIX + "ip:history:statistics";

    // ================================ 第三方服务缓存 ================================
    
    /**
     * 第三方登录状态缓存键前缀
     * 格式: poetize:oauth:state:{state}
     */
    public static final String OAUTH_STATE_PREFIX = CACHE_PREFIX + "oauth:state:";
    
    /**
     * 翻译缓存键前缀
     * 格式: poetize:translate:{hash}
     */
    public static final String TRANSLATE_PREFIX = CACHE_PREFIX + "translate:";
    
    // ================================ SEO相关缓存 ================================
    
    /**
     * Sitemap缓存键
     */
    public static final String SITEMAP_KEY = CACHE_PREFIX + "seo:sitemap";
    
    /**
     * Sitemap过期时间（秒）- 1小时
     */
    public static final long SITEMAP_EXPIRE_TIME = 3600;
    
    /**
     * 搜索引擎推送结果缓存键
     */
    public static final String SEARCH_ENGINE_PING_RESULT_KEY = CACHE_PREFIX + "seo:ping:result";
    
    /**
     * 搜索引擎推送结果缓存过期时间（秒）- 6小时
     */
    public static final long SEARCH_ENGINE_PING_RESULT_EXPIRE_TIME = 21600;
    
    // ================================ 二维码相关缓存 ================================
    
    /**
     * 文章二维码缓存键前缀
     * 格式: poetize:qrcode:article:{articleId}
     * 说明：存储文章分享二维码的字节数组，避免重复生成
     */
    public static final String ARTICLE_QRCODE_PREFIX = CACHE_PREFIX + "qrcode:article:";
    
    /**
     * 二维码缓存过期时间（秒）- 永久缓存
     * 说明：二维码内容基于文章ID固定不变，仅在文章更新/删除时主动清理缓存
     */
    public static final long QRCODE_EXPIRE_TIME = 0;

    // ================================ 缓存过期时间常量 ================================
    
    /**
     * 默认缓存过期时间（秒）- 10分钟
     */
    public static final long DEFAULT_EXPIRE_TIME = 600;
    
    /**
     * 短期缓存过期时间（秒）- 5分钟
     */
    public static final long SHORT_EXPIRE_TIME = 300;
    
    /**
     * 长期缓存过期时间（秒）- 1小时
     */
    public static final long LONG_EXPIRE_TIME = 3600;
    
    /**
     * 超长期缓存过期时间（秒）- 24小时
     */
    public static final long VERY_LONG_EXPIRE_TIME = 86400;
    
    /**
     * 永久缓存标识（秒）- 0表示永不过期
     * 用于系统核心配置等需要永久保存的缓存
     */
    public static final long PERMANENT_EXPIRE_TIME = 0;
    
    /**
     * 用户会话过期时间（秒）- 7天
     * 注意：此值应与CommonConst.TOKEN_EXPIRE保持一致，避免认证状态不同步
     * 建议使用CommonConst.TOKEN_EXPIRE替代此常量
     */
    public static final long SESSION_EXPIRE_TIME = 604800;
    
    /**
     * 验证码过期时间（秒）- 5分钟
     */
    public static final long CAPTCHA_EXPIRE_TIME = 300;
    
    /**
     * IP攻击记录过期时间（秒）- 1小时
     */
    public static final long IP_ATTACK_EXPIRE_TIME = 3600;
    
    /**
     * IP黑名单过期时间（秒）- 24小时
     */
    public static final long IP_BLACKLIST_EXPIRE_TIME = 86400;

    // ================================ 工具方法 ================================
    
    /**
     * 构建用户缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildUserKey(Integer userId) {
        return USER_CACHE_PREFIX + userId;
    }
    
    /**
     * 构建用户会话缓存键
     * @param token 会话令牌
     * @return 缓存键
     */
    public static String buildSessionKey(String token) {
        return USER_SESSION_PREFIX + token;
    }
    
    /**
     * 构建文章缓存键
     * @param articleId 文章ID
     * @return 缓存键
     */
    public static String buildArticleKey(Integer articleId) {
        return ARTICLE_CACHE_PREFIX + articleId;
    }
    
    /**
     * 构建文章列表缓存键
     * @param sortId 分类ID
     * @param page 页码
     * @param size 页大小
     * @return 缓存键
     */
    public static String buildArticleListKey(Integer sortId, Integer page, Integer size) {
        return ARTICLE_LIST_PREFIX + sortId + ":" + page + ":" + size;
    }
    
    /**
     * 构建评论列表缓存键
     * @param source 来源ID
     * @param type 类型
     * @return 缓存键
     */
    public static String buildCommentListKey(Integer source, String type) {
        return COMMENT_LIST_PREFIX + source + ":" + type;
    }
    
    /**
     * 构建IP攻击缓存键
     * @param ip IP地址
     * @return 缓存键
     */
    public static String buildIpAttackKey(String ip) {
        return IP_ATTACK_PREFIX + ip;
    }
    
    /**
     * 构建IP黑名单缓存键
     * @param ip IP地址
     * @return 缓存键
     */
    public static String buildIpBlacklistKey(String ip) {
        return IP_BLACKLIST_PREFIX + ip;
    }
    
    /**
     * 构建系统配置缓存键
     * @param configKey 配置键
     * @return 缓存键
     */
    public static String buildSysConfigKey(String configKey) {
        return SYS_CONFIG_PREFIX + configKey;
    }

    /**
     * 构建用户文章列表缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildUserArticleListKey(Integer userId) {
        return USER_ARTICLE_LIST_PREFIX + userId;
    }

    /**
     * 构建文章搜索缓存键
     * @param searchText 搜索文本
     * @return 缓存键
     */
    public static String buildSearchArticleKey(String searchText) {
        return SEARCH_ARTICLE_PREFIX + (searchText != null ? searchText.hashCode() : "empty");
    }

    /**
     * 构建用户验证码缓存键
     * @param userId 用户ID
     * @param place 位置
     * @param flag 标志
     * @return 缓存键
     */
    public static String buildUserCodeKey(Integer userId, String place, String flag) {
        return USER_CODE_PREFIX + userId + ":" + place + ":" + flag;
    }

    /**
     * 构建邮箱验证码发送次数缓存键
     * @param email 邮箱
     * @return 缓存键
     */
    public static String buildCodeMailCountKey(String email) {
        return CODE_MAIL_COUNT_PREFIX + email;
    }

    /**
     * 构建忘记密码验证码缓存键
     * @param contact 联系方式（邮箱或手机号）
     * @param flag 标志（1-手机号，2-邮箱）
     * @return 缓存键
     */
    public static String buildForgetPasswordKey(String contact, String flag) {
        return FORGET_PASSWORD_PREFIX + contact + ":" + flag;
    }

    /**
     * 构建登录失败尝试缓存键
     * @param account 账号
     * @return 缓存键
     */
    public static String buildLoginAttemptKey(String account) {
        return LOGIN_ATTEMPT_PREFIX + account;
    }

    /**
     * 构建用户保存频率限制缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildSaveCountUserKey(Integer userId) {
        return SAVE_COUNT_USER_PREFIX + userId;
    }

    /**
     * 构建IP保存频率限制缓存键
     * @param ip IP地址
     * @return 缓存键
     */
    public static String buildSaveCountIpKey(String ip) {
        return SAVE_COUNT_IP_PREFIX + ip;
    }

    /**
     * 构建用户文件上传频率限制缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildFileUploadCountUserKey(Integer userId) {
        return FILE_UPLOAD_COUNT_USER_PREFIX + userId;
    }

    /**
     * 构建IP文件上传频率限制缓存键
     * @param ip IP地址
     * @return 缓存键
     */
    public static String buildFileUploadCountIpKey(String ip) {
        return FILE_UPLOAD_COUNT_IP_PREFIX + ip;
    }

    /**
     * 构建管理员token缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildAdminTokenKey(Integer userId) {
        return ADMIN_TOKEN_PREFIX + userId;
    }

    /**
     * 构建用户token缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildUserTokenKey(Integer userId) {
        return USER_TOKEN_PREFIX + userId;
    }

    /**
     * 构建管理员token间隔检查缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildAdminTokenIntervalKey(Integer userId) {
        return ADMIN_TOKEN_INTERVAL_PREFIX + userId;
    }

    /**
     * 构建用户token间隔检查缓存键
     * @param userId 用户ID
     * @return 缓存键
     */
    public static String buildUserTokenIntervalKey(Integer userId) {
        return USER_TOKEN_INTERVAL_PREFIX + userId;
    }
    
    // ================================ 访问统计缓存键构建方法 ================================
    
    /**
     * 构建今日访问计数缓存键
     * @param date 日期（格式：yyyy-MM-dd）
     * @return 缓存键
     */
    public static String buildTodayVisitCountKey(String date) {
        return TODAY_VISIT_COUNT_PREFIX + date;
    }
    
    /**
     * 构建每日访问记录缓存键
     * @param date 日期（格式：yyyy-MM-dd）
     * @return 缓存键
     */
    public static String buildDailyVisitRecordsKey(String date) {
        return DAILY_VISIT_RECORDS_PREFIX + date;
    }
    
    /**
     * 构建IP今日访问标记缓存键
     * @param date 日期（格式：yyyy-MM-dd）
     * @param ip IP地址
     * @param userId 用户ID（可为null）
     * @return 缓存键
     */
    public static String buildIpTodayVisitKey(String date, String ip, Integer userId) {
        String userSuffix = userId != null ? "_" + userId : "";
        return IP_TODAY_VISIT_PREFIX + date + ":" + ip + userSuffix;
    }
    
    // ================================ 二维码缓存键构建方法 ================================
    
    /**
     * 构建文章二维码缓存键
     * @param articleId 文章ID
     * @return 缓存键
     */
    public static String buildArticleQRCodeKey(Integer articleId) {
        return ARTICLE_QRCODE_PREFIX + articleId;
    }
}
